chrome/content/zotero/xpcom/attachments.js

```diff
@@ -332,6 +332,17 @@ Zotero.Attachments = new function(){
 					+ "'application/pdf' in Attachments.importFromURL()";
 				Zotero.debug(errString, 2);
 				Zotero.debug(sample, 3);
+				try {
+					if (tmpDir && tmpDir.exists()) {
+						tmpDir.remove(true);
+					}
+					if (destDir && destDir.exists()) {
+						destDir.remove(true);
+					}
+				}
+				catch (e) {
+					Zotero.debug(e, 1);
+				}
```

Comment:
Not entirely sure what you mean here. We can only check the content type of the content after it is saved, so cleanup in `Zotero.Utilities.Internal.saveURI` doesn't really work, unless we do the content type check there (which I don't think we should).